src/rcv/septentrio.c

```diff
@@ -119,802 +337,2401 @@ static uint8_t sig_tbl[SBF_MAXSIG+1][2]={ /* system, obs-code */
+    uint8_t *p=(raw->buff)+6;
+
+
+    trace(4,"SBF decode_sbsfastcorrdegr:\n");
+
+    if (raw->len<68)
+    {
+        trace(1,"SBF decode_sbsfastcorrdegr: Block too short\n");
+        return -1;
+    }
+
+    /* get satellite number */
+    prn=U1(p+8);
+    if (prn < 120) return -1;
+    if (prn > 139) return -1;
+
+    if (raw->outtype) {
+        sprintf(raw->msgtype,"SBF SBAS Fast Correction Degradation Factor from PRN=%d", prn);
+    }
+
+    if (raw->nav.sbssat.iodp!=U1(p+9)) return 0;
+
+    raw->nav.sbssat.tlat=U1(p+10);
+
+    for (i=0;i<raw->nav.sbssat.nsat&&i<MAXSAT;i++) {
+        raw->nav.sbssat.sat[i].fcorr.ai=U1(p+11+i);
+    }
+    return 0;
+}
+
+/* decode type 26: ionospheric delay corrections -----------------------------*/
+static int decode_sbsionodelay(raw_t *raw)
+{
+    int i, j, give, prn;
+    int band;
+    uint8_t *p = (raw->buff)+8, sbLength, count;
+    uint16_t week;
+    uint32_t tow;
+
+    trace(4, "SBF decode_sbsionodelay:\n");
+
+    if (raw->len<20)
+    {
+        trace(1, "SBF decode_sbsionodelay: Block too short\n");
+        return -1;
+    }
+
+    /* get satellite number */
+    prn = U1(p+6);
+    if (prn < 120) return -1;
+    if (prn > 139) return -1;
+
+    if (raw->outtype) {
+        sprintf(raw->msgtype, "SBF SBAS Ionospheric Delay Correction from PRN=%d", prn);
+    }
+
+    band = U1(p+7);
+
+    if (band>MAXBAND || raw->nav.sbsion[band].iodi != U1(p+8)) return 0;
+
+    tow = U4(p)/1000;
+    week = U2(p+4);
+
+    count = U1(p+9);
+    sbLength = U1(p+10);
+
+    if (count != 15)
+    {
+        trace(1, "SBF decode_sbsionodelay: wrong number of IDC blocks: %d\n", count);
+        return -1;
+    }
+
+    for (i=0; i<count; i++) {
+        j = U1(p+12+i*sbLength);
+        give = U1(p+12+i*sbLength+1);
+
+        raw->nav.sbsion[band].igp[j].t0 = gpst2time(week, tow);
+        raw->nav.sbsion[band].igp[j].delay = R4(p+12+i*sbLength+4);
+        raw->nav.sbsion[band].igp[j].give = give;
+
+        if (raw->nav.sbsion[band].igp[j].give >= 16) {
+            raw->nav.sbsion[band].igp[j].give = 0;
+        }
+    }
+    trace(5, "decode_sbsionodelay: band=%d\n", band);
+    return 3;
+}
+
+/* decode type 18: ionospheric grid point masks ------------------------------*/
+static int decode_sbsigpmask(raw_t *raw) /* TODO: verify this function */
+{
+    const sbsigpband_t *b;
+    int i, j, n, m, prn;
+    uint8_t band;
+
+    uint8_t *p = (raw->buff)+8;
+
+    trace(4, "SBF decode_sbsigpmask:\n");
+
+    if (raw->len < 20)
+    {
+        trace(1, "SBF decode_sbsigpmask: Block too short\n");
+        return -1;
+    }
+
+    /* get satellite number */
+    prn = U1(p+6);
+    if (prn < 120) return -1;
+    if (prn > 139) return -1;
+
+    if (raw->outtype) {
+        sprintf(raw->msgtype, "SBF SBAS Ionospheric Grid Points from PRN=%d", prn);
+    }
+
+    band = U1(p+8);
+
+    if      (band <= 8) {b = igpband1[band]; m = 8;}
+    else if (9<=band && band<=10) {b = igpband2[band-9]; m = 5;}
```

Comment:
9<=band is unnecessary, due to above test.

Reply:
thanks